app/services/user_signup.rb

```diff
@@ -0,0 +1,56 @@
+class UserSignup
+  attr_reader :error_message
+
+  def initialize(user)
+    @user = user
+  end
+
+  def sign_up(stripeToken, invitation_token)
+    charge = StripeWrapper::Charge.create(
+      :amount => 999,
+      :source => stripeToken,
+      :description => "Monthly Subscription for @{@user.email_address}"
+    )
+    if @user.valid?
+      if charge.successful?
+        @user.save
+        handle_invitation(invitation_token)
+        AppMailer.welcome_new_user(@user).deliver
+        @status = :success
+        self
+      else
+        @status = :fail
+        @error_message = charge.error_message
+        self
+      end
+    else
+      @status = :fail
+      @error_message = "Please check input fields."
+      self
```

Comment:
you can refactor `self` at the end of this method and delete on lines 20, 24, 29.